### PR TITLE
add event_expansion for v1 Events

### PR DESCRIPTION
**Pull request by cndoit18**
Signed-off-by: cndoit18 [cndoit18@outlook.com](mailto:cndoit18@outlook.com)
 
**What type of PR is this?**

/kind feature
/sig scalability
 
**What this PR does / why we need it**:

We want to find its events by object reference, and now the client-go has no expansion method.
this PR adds an expansion to client-go.

**Which issue(s) this PR fixes**:
 
**Special notes for your reviewer**:
 
**Does this PR introduce a user-facing change?**:
 
```release-note
NONE
```

### CODE REVIEW

**Comment by cndoit18** [comment]
sig/scalability

File: staging/src/k8s.io/client-go/kubernetes/typed/events/v1/event_expansion.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+type EventExpansion interface {
+	// CreateWithEventNamespace is the same as a Create
+	// except that it sends the request to the event.Namespace.
+	CreateWithEventNamespace(event *eventsv1.Event) (*eventsv1.Event, error)
+	// UpdateWithEventNamespace is the same as a Update
+	// except that it sends the request to the event.Namespace.
+	UpdateWithEventNamespace(event *eventsv1.Event) (*eventsv1.Event, error)
+	// PatchWithEventNamespace is the same as an Update
+	// except that it sends the request to the event.Namespace.
+	PatchWithEventNamespace(event *eventsv1.Event, data []byte) (*eventsv1.Event, error)
```
Comment (liggitt) [inline staging/src/k8s.io/client-go/kubernetes/typed/events/v1/event_expansion.go:41]: using the standard create/update/patch methods seems equivalent and preferable

**Comment by liggitt** [comment]
these were proposed in https://github.com/kubernetes/kubernetes/pull/83065, and the recommendation was to make these helper methods that take and events client, rather than making them part of the events clients interface

**Comment by cndoit18** [comment]
> these were proposed in #83065, and the recommendation was to make these helper methods that take and events client, rather than making them part of the events clients interface

Thank you, I read this PR, it is useful, but it has been closed.

**Comment by cndoit18** [comment]
I plan to re-write the code according to this idea. @liggitt

**Pull request merged**